File: src/oldSnacks/CreditCard.java
package oldSnacks;

import java.util.Scanner;

public class CreditCard {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        CreditCardMethod cc = new CreditCardMethod();
        boolean validate = false;
        System.out.println("App to validate CreditCard");
        System.out.print("Hello, Kindly enter card number to verify: ");
        String number = input.nextLine();
        while (!cc.validateStringOfNumbers(number) || !cc.validateLength(number)){
            System.out.print("Please make sure card number is correct, try again: ");
            number = input.nextLine();
        }

        int[] array = cc.convertToArray(number);
        int sumOne = cc.sumofArrayFromLastIndex(array);
        int sumTwo = cc.sumofArrayLuhmPrincipleIndex(array);
        int total = sumOne + sumTwo;

        if (total % 10 == 0) {
            validate = true;
        }

        String result = validate ? "valid" : "inValid";

        if (result == "valid") {

            System.out.printf("""
                    *****Credit Card Type: %s
                    *****Credit Card number: %s
                    *****Credit Card Digit Length: %d
                    *****Credit Card Validity Status: %s
                    """, cc.getCardType(number), number, number.length(), result);
            System.out.println("Thank you for using the Credit Card Validator App!");
        }
        else{
            System.out.printf("""
                    *****Credit Card Type: Invalid Card
                    *****Credit Card number: %s
                    *****Credit Card Digit Length: %d
                    *****Credit Card Validity Status: %s
                    """,  number, number.length(), result);
            System.out.println("Please try again with a valid card number.");
            System.out.println("Thank you for using the Credit Card Validator App!");
        }






    }
}

File: JavaCript for PhaseGate/CreditCard.js
const {
    convertToArray,
    validateLength,
    sumOfArrayAtLastIndex,
    sumOfArrayAtFirstIndex,
    getCardType,
    validateStringOfNumber} = require('./CreditCardMethod.js');


const prompt = require('prompt-sync')();


console.log("App to validate CreditCard");
let creditCardNumber = prompt("Please enter the credit card number: ");
while(!validateStringOfNumber(creditCardNumber) || !validateLength(creditCardNumber)){
    console.log("Wrong card input, try again: ");
    creditCardNumber = prompt();
}

array = convertToArray(creditCardNumber);
sumOne = sumOfArrayAtLastIndex(array);
sumTwo = sumOfArrayAtFirstIndex(array);
total = sumOne + sumTwo;

validate = false;

if (total % 10 == 0) {
    validate = true;
}

result = validate ? "valid" : "inValid";

if (result == "valid") {
    const text = `
****Credit Card Type: ${getCardType(creditCardNumber)}
****Credit Card number: ${creditCardNumber}
****Credit Card Digit Length: ${creditCardNumber.length}
****Credit Card Validity Status: ${result}
    `;

    console.log(text);
}
else{
    const text = `
*****Credit Card Type: Invalid Card
*****Credit Card number: ${creditCardNumber}
*****Credit Card Digit Length: ${creditCardNumber.length}
*****Credit Card Validity Status: ${result}
            `;
    console.log(text);
    console.log("Please try again with a valid card number.");
    console.log("Thank you for using the Credit Card Validator App!");
}

